adoc/chapters/programming_interface.adoc

```diff
@@ -5186,718 +5173,636 @@ If no template argument is provided, then the default allocator for the SYCL
+The buffer assumes exclusive access to this memory for the duration of its
+lifetime.
 
-Zero or more properties can be provided to the constructed SYCL
-[code]#buffer# via an instance of
-[code]#property_list#.
-
-a@
-[source]
-----
-buffer(const T* hostData,
-       const range<Dimensions>& bufferRange,
-       AllocatorT allocator,
-       const property_list& propList = {})
-----
-   a@ Construct a SYCL [code]#buffer# instance with the [code]#hostData#
-      parameter provided.  The buffer assumes exclusive access to this
-      memory for the duration of its lifetime.
-
-The constructed SYCL [code]#buffer# will use the
-[code]#allocator# parameter provided when allocating
-memory on the host.
-
-The host address is [code]#const T#, so the host accesses
-can be read-only. However, the [code]#typename T# is not const so
-the device accesses can be both read and write accesses. Since,
-the [code]#hostData# is const, this buffer is only initialized
-with this memory and there is no write back after its
-destruction, unless the [code]#buffer# has another valid
-non-null final data address specified via the member function
-[code]#set_final_data()# after construction of the
+The constructed SYCL [code]#buffer# will use the [code]#allocator# parameter
+provided when allocating memory on the host.
+
+The host address is [code]#const T#, so the host accesses can be read-only.
+However, the [code]#typename T# is not const so the device accesses can be both
+read and write accesses.
+Since, the [code]#hostData# is const, this buffer is only initialized with this
+memory and there is no write back after its destruction, unless the
+[code]#buffer# has another valid non-null final data address specified via the
+member function [code]#set_final_data()# after construction of the
 [code]#buffer#.
 
-The range of the constructed SYCL [code]#buffer# is
-specified by the [code]#bufferRange# parameter provided.
+The range of the constructed SYCL [code]#buffer# is specified by the
+[code]#bufferRange# parameter provided.
 
-Zero or more properties can be provided to the constructed SYCL
-[code]#buffer# via an instance of
-[code]#property_list#.
+Zero or more properties can be provided to the constructed SYCL [code]#buffer#
+via an instance of [code]#property_list#.
 
+'''
 
-a@
-[source]
+.[apititle]#Construct from a container#
+[source,role=synopsis,id=api:buffer-ctor-container]
 ----
 template <typename Container>
-buffer(Container& container,
-       const property_list& propList = {})
-----
-   a@ Construct a one dimensional SYCL [code]#buffer# instance
-      from the elements starting at [code]#std::data(container)#
-      and containing [code]#std::size(container)# number of elements.
-      The buffer is initialized with the contents of [code]#container#,
-      and the buffer assumes exclusive access to [code]#container# for
-      the duration of its lifetime.
+buffer(Container& container, AllocatorT allocator,                 (1)
```

Comment:
> Thanks, @gmlueck! I believe I've addressed most of the initial comments, but I'm still a little fuzzy on the placement of the callouts.

The callouts mostly look good, except they should be moved to the first line of each function definition.  Using this one as an example, move the "(1)" one line earlier, so that it's on the same line as the `template` keyword.  The justification should be the same as what you have now.  For example, this "(1)" starts in column 68.  It should still be in column 68 when you move it one line earlier.